### PR TITLE
🌱 Bump to controller-runtime v0.14.5

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
This PR picks up a new CR version which includes a few useful metrics for us:
* https://github.com/kubernetes-sigs/controller-runtime/pull/2217

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by sbueringer** [comment]
/hold 
waiting for CR release + make generate-modules

File: hack/observability/prometheus/values.yaml [hunk 1]
```diff
@@ -14,6 +14,8 @@ pushgateway:
 # Scrape metrics from deployed providers
 extraScrapeConfigs: |
   - job_name: 'capi-providers'
+    # 15s is a bit often for production but helps to get metrics quicker for development.
```
Comment (jackfrancis) [inline hack/observability/prometheus/values.yaml:17]: rebase errata?

Comment (sbueringer) [reply hack/observability/prometheus/values.yaml:17]: The idea was to make the metrics that we pick up with CR v0.14.5 better accessible in the dev env.

But I can move it to a separate PR if we prefer

Comment (jackfrancis) [reply hack/observability/prometheus/values.yaml:17]: got it, IMO including this change in this PR makes sense

Comment (sbueringer) [reply hack/observability/prometheus/values.yaml:17]: Okay. I'm really fine either way

**Comment by sbueringer** [comment]
A bit of context: ~ waiting for this PR: https://github.com/kubernetes-sigs/controller-runtime/pull/2219

**Comment by sbueringer** [comment]
Should be ready now

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Pull request merged**